From b1925829e6242eafd2d775894c312be75e7676fc Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Tue, 28 Mar 2017 01:37:02 +0800
Subject: [PATCH 453/637] drm/i915/gvt: return correct PCH type in GVTg guest
 of BXT

In GVT environment on BXT platform, intel_detect_pch will return PCH_NOP,
which will impact the i915 display logic in GVTg guest. This patch is to
fix this issue.

V2: uses a more elegant way to assign correct PCH type

Change-Id: I6fa13f78ca10ae2d5b41cd43fd0902559f68ba16
Signed-off-by: Min He <min.he@intel.com>
Reviewed-on: https://localhost/613350
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 82a6a94a120e..2220601b84d1 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -146,6 +146,9 @@ static enum intel_pch intel_virt_detect_pch(struct drm_i915_private *dev_priv)
 	} else if (IS_COFFEELAKE(dev_priv) || IS_CANNONLAKE(dev_priv)) {
 		ret = PCH_CNP;
 		DRM_DEBUG_KMS("Assuming CannonPoint PCH\n");
+	} else if (IS_BROXTON(dev_priv)) {
+		ret = PCH_NONE;
+		DRM_DEBUG_KMS("Assuming None PCH for BXT\n");
 	}
 
 	return ret;
-- 
2.16.2

